astropy/cosmology/core.py

```diff
@@ -166,23 +335,96 @@ def __eq__(self, other):
         Returns
         -------
         bool
-            True if all immutable fields are equal, False otherwise.
+            True if Parameters and names are the same, False otherwise.
+        """
+        param_eq = Cosmology.equivalent(self, other)  # (also False if diff class)
```

Comment:
Why not `self.equivalent(other)`?

Reply:
Because subclasses define methods ``equivalent`` that mean 2 different classes can be equivalent. The ``Cosmology.equivalent`` defines a strict equality of the parameters. It's useful for not duplicating code between the various ``equivalent`` and ``__eq__``.